src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLanguageServerClient.cs

```diff
@@ -41,29 +42,17 @@ internal class RazorLanguageServerClient : ILanguageClient
 
         public async Task<Connection> ActivateAsync(CancellationToken token)
         {
-            await Task.Yield();
-
-            var currentAssembly = typeof(RazorLanguageServerClient).Assembly;
-            var currentAssemblyLocation = currentAssembly.Location;
-            var extensionDirectory = Path.GetDirectoryName(currentAssemblyLocation);
-            var languageServerPath = Path.Combine(extensionDirectory, "LanguageServer", "rzls.exe");
-            var info = new ProcessStartInfo();
-            info.FileName = languageServerPath;
-            info.Arguments = "-lsp --trace Verbose";
-            info.RedirectStandardInput = true;
-            info.RedirectStandardOutput = true;
-            info.UseShellExecute = false;
-            info.CreateNoWindow = true;
-
-            Process process = new Process();
-            process.StartInfo = info;
-
-            if (process.Start())
-            {
-                return new Connection(process.StandardOutput.BaseStream, process.StandardInput.BaseStream);
-            }
+            var (clientStream, serverStream) = FullDuplexStream.CreatePair();
+
+            // Need an auto-flushing stream for the server because O# doesn't currently flush after writing responses. Without this
```

Comment:
This sounds like a thing that was learned through a long painful process lol

Reply:
You have no idea